lib/IRGen/GenType.cpp

```diff
@@ -1140,6 +1146,31 @@ static bool doesPlatformUseLegacyLayouts(StringRef platformName,
 
   return false;
 }
+
+// The following Apple platforms ship an Objective-C runtime supporting
+// the class metadata update hook:
+//
+// - macOS 10.14.4
+// - iOS 12.2
+// - tvOS 12.2
+// - watchOS 5.2
+static bool doesPlatformSupportObjCMetadataUpdateCallback(
+    const llvm::Triple &triple) {
+  if (triple.isMacOSX())
+    if (!triple.isMacOSXVersionLT(10, 14, 4))
+      return true;
+
+  if (triple.isiOS()) // also returns true on tvOS
+    if (!triple.isOSVersionLT(12, 2))
+      return true;
+
+  if (triple.isWatchOS())
+    if (!triple.isOSVersionLT(5, 2))
+      return true;
```

Comment:
Nitpick: these should all just return the value instead of cascading in the `else` branch.

Reply:
Refactored. Also, what's an "else" branch? The LLVM coding style prohibits 'if' statements without an early return :)